pyoutline/tests/layer_test.py

```diff
@@ -0,0 +1,342 @@
+        """Test get/set of chunk size."""
+        self.event.set_chunk_size(5)
+        self.assertEquals(5, self.event.get_chunk_size())
+
+    def test_add_layer_during_setup(self):
+        """Test to ensure that layers added during setup are serialized."""
+        with TemporarySessionDirectory():
+            ol = outline.Outline('mr_hat')
+            ol.add_layer(TestA('test_a'))
+            ol.setup()
+
+            ol_b = outline.load_outline(ol.get_session().get_file('outline.yaml'))
+            # ensure the new layer was added
+            self.assertTrue(ol_b.get_layer('test_b'))
+            #  ensure that setup was actually run on the new layer
+            self.assertTrue(ol_b.get_layer('test_b').is_setup)
+
+    def test_after_init(self):
+        ol = outline.Outline('after_init')
+        ol.add_layer(TestAfterInit('test'))
+
+        # Ensure after init was run,
+        self.assertTrue(ol.get_layer('test').get_arg('after_init'))
+        # Ensure that the layer has the right ol reference
+        self.assertEquals(ol, ol.get_layer('test').get_outline())
+
+    def test_after_init_current(self):
+        ol = outline.Outline('after_init', current=True)
+        TestAfterInit('test')
+
+        # Ensure after init was run,
+        self.assertTrue(ol.get_layer('test').get_arg('after_init'))
+        # Ensure that the layer has the right ol reference
+        self.assertEquals(ol, ol.get_layer('test').get_outline())
+
+    def test_dependency_creation(self):
+        with TemporarySessionDirectory():
+            outline.Outline.current = None
+            ol = outline.Outline('after_init')
+            ol.add_layer(TestAfterInit('test'))
+            ol.add_layer(TestB('testb', require='test'))
+            ol.setup()
+
+            # check the depend was setup properly
+            self.assertEquals(1, len(ol.get_layer('testb').get_depends()))
+
+    def test_type_arg(self):
+        """Test to ensure the type argument is handled properly."""
+        outline.Outline.current = None
+        t = TestA('test', type='Post')
+        self.assertEquals('Post', t.get_type())
+
+    def test_set_output_attribute(self):
+        """Test setting an output attribute on all registered output."""
+        t = TestA('test')
+        t.add_output('node1', outline.io.Path('/tmp'))
+        t.set_output_attribute('test', True)
+        self.assertTrue(t.get_output('node1').get_attribute('test'))
+
+    def test_set_output_attribute(self):
+        """Test setting an input attribute on all registered input."""
+        t = TestA('test')
+        t.add_input('node1', outline.io.Path('/tmp'))
+        t.set_input_attribute('test', True)
+        self.assertTrue(t.get_input('node1').get_attribute('test'))
+
+class OutputRegistrationTest(unittest.TestCase):
+
+    def setUp(self):
+        outline.Outline.current = None
+
+    # TODO(bcipriano) Re-enable this test once FileSequence has a Python implementation.
```

Comment:
Do we have an issue number for this?

Reply:
Ah yeah, meant to add that earlier. Good catch.

https://github.com/imageworks/OpenCue/issues/242